include/mata/nfa.hh

```diff
@@ -804,7 +809,20 @@ public:
      * The value of the already existing symbols will NOT be overwritten.
      */
     void add_symbols_to(OnTheFlyAlphabet& alphabet);
-}; // Nfa
+}; // struct Nfa.
+
+/**
+ * Create automaton accepting only epsilon string.
+ */
+[[nodiscard]] Nfa create_empty_string_nfa();
```

Comment:
what is `[[nodiscard]]`? Something new from C++17? I would advice you to discuss using these new fancy things @kilohsakul. Just to make sure that he likes them. You would save your time removing them later ;-)

Reply:
It is a C++17 attribute, which tells the compiler to issue a warning if the user calls this function from a discarded-value expressions; i.e., when the function is called yet the return value is discarded and not used. It should be used for all functions without any side effects (without any out parameters, function that are not setters, etc.).

The user should know that when they want to use this function, they must somehow make use of the return value, otherwise the call to the function has no effect.

Linters suggest to use `[[nodiscard]]` whenever possible these days.

Reply:
Do not use.